airbyte-webapp-e2e-tests/cypress/pages/replicationPage.ts

```diff
@@ -74,29 +73,90 @@ export const selectSyncMode = (source: string, dest: string) => {
   cy.get(`.react-select__option`).contains(`Source:${source}|Dest:${dest}`).click();
 };
 
-export const selectCursorField = (value: string) => {
-  cy.get(cursorFieldDropdown).first().click({ force: true });
-
-  cy.get(`.react-select__option`).contains(value).click();
+type Dropdown = "cursor" | "primaryKey";
+/**
+ * General function - select dropdown option(s)
+ * @param streamName
+ * @param dropdownType
+ * @param value
+ */
+const selectFieldDropdownOption = (streamName: string, dropdownType: Dropdown, value: string | string[]) => {
+  const container = getFieldDropdownContainer(streamName, dropdownType);
+  const button = getFieldDropdownButton(streamName, dropdownType);
+
+  cy.get(container).within(() => {
+    cy.get(button).click();
+
+    if (Array.isArray(value)) {
+      // in case if multiple options need to be selected
+      value.forEach((v) => cy.get(getFieldDropdownOption(v)).click());
+    } else {
+      // in case if one option need to be selected
+      cy.get(getFieldDropdownOption(value)).click();
+    }
+  });
+  // close dropdown - click on dropdown overlay if exist
+  cy.get("body").then(($body) => {
+    if ($body.find(dropDownOverlayContainer).length > 0) {
+      cy.get(dropDownOverlayContainer).click();
+    }
+  });
 };
 
-export const checkStreamFields = (listNames: string[], listTypes: string[]) => {
-  cy.get(streamFieldNames).each(($span, i) => {
+/**
+ * Select cursor value from cursor dropdown(pathPopout) in desired stream
+ * @param streamName
+ * @param cursorValue
+ */
+export const selectCursorField = (streamName: string, cursorValue: string) =>
```

Comment:
Duly noted! Agree! One of the possible solutions: 
1. In order to avoid cases with the same `streamName`(across different namespaces), before moving to the next stream, we can collapse the previous one.
2. Extend the `selectCursorField` to support namespace. (But I'm not sure this will live long since in the new design, stream details opens in new window)

WDYT?

Reply:
Great point that this will not be an issue once the new panel is live, and that it is only an issue if they both happen to be expanded as well!  I think just adding a cleanup that collapses it should be fine (no strong feelings on that either, since we do know the schema because we're creating it ourselves)

Reply:
Yeap, cleaning function would be enough atm 😃